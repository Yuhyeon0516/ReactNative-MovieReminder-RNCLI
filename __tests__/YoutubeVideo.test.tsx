import React from 'react';
import YouTubeVideo from '../src/screens/Detail/YouTubeVideo';
import renderer from 'react-test-renderer';
import {Linking, TouchableOpacity} from 'react-native';

describe('<YoutubeVideco />', () => {
    const youTubeKey = 'test_key';
    let testRenderer: renderer.ReactTestRenderer;

    beforeEach(() => {
        testRenderer = renderer.create(
            <YouTubeVideo title="Test Title" youTubeKey={youTubeKey} />,
        );
    });

    it('renders correctly', () => {
        const snapShot = testRenderer.toJSON();
        expect(snapShot).toMatchSnapshot();
    });

    it('open url when it is pressed', () => {
        const spyFn = jest.spyOn(Linking, 'openURL');
        const touchableOpacity = testRenderer.root.findByType(TouchableOpacity);

        touchableOpacity.props.onPress();

        expect(spyFn).toHaveBeenCalledWith(
            `https://www.youtube.com/watch?v=${youTubeKey}`,
        );
    });
});
