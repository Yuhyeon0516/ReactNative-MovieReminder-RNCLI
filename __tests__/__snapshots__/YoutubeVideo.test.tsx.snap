// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<YoutubeVideco /> renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onLayout={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignSelf": "stretch",
      "backgroundColor": "#ffffff",
      "borderRadius": 12,
      "opacity": 1,
      "overflow": "hidden",
    }
  }
>
  <Image
    source={
      {
        "uri": "https://img.youtube.com/vi/test_key/0.jpg",
      }
    }
    style={
      {
        "height": 0,
      }
    }
  />
  <Text
    style={
      {
        "color": "#000000",
        "fontSize": 16,
        "padding": 12,
      }
    }
  >
    Test Title
  </Text>
</View>
`;
